Add tests, integrate with CI

Rastopyr opened the pull request:
Hi @yoavlt,

i'm added few tests and integrations with travis and coveralls. This is preliminary changes before extend API of this library. For me is very matter, because i needed for work with Youtube.

Regards,
Roman

NOTE: If you will approved this PR, i will send few next, with improved test coverage and extended API(channels for example).

Rastopyr committed:
0.0.8

* Add general tests for public methods
* Add ex.vcr for fixture of Youtube API

Rastopyr committed:
0.0.9

* Add detailed documentations
* Little bit fixes of tests

yoavlt commented:
Thank you for your PR ! 👍 
but, do not contain your youtube key and your converall ID.
I can't merge your pull request.

Rastopyr committed:
Remove secured keys

Rastopyr commented:
@yoavlt sorry about that. my mistake. already fixed :)

Rastopyr committed:
[0.0.10] Big patch.

* Improve test coverage
  * `Tubex.Video`.
  * `Tubex.API`. Actually, only stub tests for prevent regression
  * `Tubex.Utils`.
* Add spec for `Tubex.Video`
* Add docs for `Tubex.Video`
* Separate `Tubex.API.get` and `page_info` object in `Tubex.Video` to
independent function
* Update `ex_doc` version

Rastopyr committed:
Resolve conflicts

Rastopyr committed:
Remove `usage` block from `Readme.md` on favor online documentation

Rastopyr commented:
@yoavlt any updates? 😸

Rastopyr committed:
Big change.

Fork of https://github.com/yoavlt/tubex.

Motivation
====

Actually, `tubex` not support of @yoavlt about 11 months.
But package is uses and downloaded each day. Maintainer of master
package not approve any pull requests and updates.

So, for future support of YouTube functionality in elixir flow,
i decided to fork `tubex` package.

Roadmap
===

Authorship belongs to @yoavlt.

1. Refactor API to more polymorphous with origin API of YouTube

For example `TubEx.Video.details` will be `TubEx.Video.get`.
`Tubex.Video.search_by_query` will be `Tubex.Video.search`.

More details will be in relates issue

2. Channel support
3. Playlist support
4. Comment support

Rastopyr committed:
Bump version and update Readme.md

Rastopyr committed:
Rename Video method.
Change tests.

Rename repo to correct name
